src/main/java/seedu/address/logic/commands/DeleteStudentCommand/DeleteStudentByIndexCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands.deletestudentcommand;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Deletes a student identified using index from the address book.
+ */
+public class DeleteStudentByIndexCommand extends DeleteStudentCommand {
+
+    private final Index targetIndex;
+
+    /**
+     * Creates a DeleteStudentCommand to delete the student with the specified
+     * {@code studentId}.
+     */
+    public DeleteStudentByIndexCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Person personToDelete;
+        personToDelete = model.getFilteredPersonList().get(targetIndex.getZeroBased());
+        model.deletePerson(personToDelete);
```

Comment:
Is there supposed to be a check for null here? for example: if the index input by user is larger than the number of students in the list.

Reply:
Similarly, if the index is negative

Reply:
Thanks for spotting this bug! Will fix it! 👍